### DISCUSSION TITLE
Can't merge document using dataset

### DISCUSSION THREAD

**Discussion by leefsmp** [donmccurdy/glTF-Transform#629] [Q&A] [answered]
I've looked at [that](https://stackoverflow.com/questions/68098555/merging-multiple-gltf-having-some-common-nodes-using-gltf-transform) SO thread and doing something pretty similar:

```
const input = [{
    name: '2213-T1-A-Z0-CL'
  }, {
    name: '2213-T1-A-Z1-CO'
  }, {
    name: '2213-T1-A-Z7-CL'
}]

const io = new NodeIO()

 let merged = new Document();

 for (let entry of input) {
   merged = merged.merge(await io.read(path.join("./tests/output", `${entry.name}.gltf`)))
 }

 await io.write(
   path.join("./tests/output", `merged.gltf`), 
   merged)
 ```
 
 However it seems to overwrite the data in gltf so I'm just seeing the 3rd document. Trying to open each gltf independently in the viewer works just fine. Is this a problem specific to my dataset (see the 3 models attached below in input.zip and result.zip)?  

Thanks in advance for any insights.

      
[input.zip](https://github.com/donmccurdy/glTF-Transform/files/9135917/input.zip)
[result.zip](https://github.com/donmccurdy/glTF-Transform/files/9135918/result.zip)

**Comment by donmccurdy** [comment, marked as answer]
The code above creates a glTF file containing three scenes — one for each input file. That's completely valid, but unusual, and viewers will just show one scene by default (perhaps with an option to change scenes somewhere). If you would like to merge them into a single scene viewers will show all at once, you'll need to add a bit more code from the SO answer:

```js
const root = merged.getRoot();
const mainScene = root.listScenes()[0];

for (const scene of root.listScenes()) {
  if (scene === mainScene) continue;

  for (const child of scene.listChildren()) {
    mainScene.addChild(child);
  }

  scene.dispose();
}
```

**Reply by leefsmp** [reply]
Thanks that's awesome. If my desired output is just a single gltf with a single scene containing all geometry, do I still need to merge each document first or I could directly `addChild` from a `Document` into the output gltf?

**Reply by donmccurdy** [reply]
You'll need to merge first — otherwise it'll throw an error when trying to add a Node (or anything else) from one Document to another Document.
